src/Features/LanguageServer/Protocol/Handler/Completion/AbstractLspCompletionResultCreationService.cs

```diff
@@ -179,6 +181,37 @@ async Task<LSP.CompletionItem> CreateLSPCompletionItemAsync(CompletionItem item,
                 return LSP.CompletionItemKind.Text;
             }
 
+            static LSP.CompletionItemTag[]? GetCompletionTags(
+                ImmutableArray<string> tags,
+                ISet<LSP.CompletionItemTag> supportedClientTags)
+            {
+                using var result = TemporaryArray<LSP.CompletionItemTag>.Empty;
+
+                foreach (var tag in tags)
+                {
+                    if (ProtocolConversions.RoslynTagToCompletionItemTags.TryGetValue(tag, out var completionItemTags))
+                    {
+                        // Always at least pick the core tag provided.
+                        var lspTag = completionItemTags[0];
+
+                        // If better kinds are preferred, return them if the client supports them.
+                        for (var i = 1; i < completionItemTags.Length; i++)
+                        {
+                            var preferredTag = completionItemTags[i];
+                            if (supportedClientTags.Contains(preferredTag))
+                                lspTag = preferredTag;
```

Comment:
Shouldn't the add be moved here in some way? Unless I'm reading it wrong, given tags on an item of `"class", "foo"` and server supported tags of `"class", "struct"`, this method would return `"class", "class"`.

Reply:
For the case you describe, this method would return `"class", "foo"`.